pkg/reconciler/common/transformer_additional_options.go

```diff
@@ -135,6 +135,66 @@ func (ot *OptionsTransformer) updateAnnotations(u *unstructured.Unstructured, an
 	return nil
 }
 
+func (ot *OptionsTransformer) updatePodTemplateLabels(u *unstructured.Unstructured, labels map[string]string) error {
+	if len(labels) == 0 {
+		return nil
+	}
+
+	actualLabels, _, err := unstructured.NestedMap(u.Object, "spec", "template", "metadata", "labels")
+
+	if err != nil {
+		return err
+	}
+
+	if actualLabels == nil {
+		actualLabels = make(map[string]interface{})
+	}
+
+	for labelKey, labelValue := range labels {
+		actualLabels[labelKey] = labelValue
+	}
+
+	if actualLabels == nil {
```

Comment:
@charlesmorin `actualLabels` never be `nil` right? You have already initialized with empty map, if it is `nil`